src/Microsoft.ML.Data/Scorers/MultiClassClassifierScorer.cs

```diff
@@ -292,9 +291,31 @@ public ISchemaBindableMapper Bindable
                     _labelNameType = type;
                     _labelNameGetter = getter;
                     _metadataKind = metadataKind;
-
-                    _outSchema = new SchemaImpl(mapper.OutputSchema, scoreIdx, _labelNameType, _labelNameGetter, _metadataKind);
                     _canWrap = canWrap;
+
+                    OutputSchema = DecorateOutputSchema(mapper.OutputSchema, scoreIdx, _labelNameType, _labelNameGetter, _metadataKind);
+                }
+
+                /// <summary>
+                /// Append label names to score column as its metadata.
+                /// </summary>
+                private Schema DecorateOutputSchema(Schema partialSchema, int scoreColumnIndex, VectorType labelNameType,
+                    ValueGetter<VBuffer<T>> labelNameGetter, string labelNameKind)
+                {
+                    var builder = new SchemaBuilder();
+                    for (int i = 0; i < partialSchema.ColumnCount; ++i)
+                    {
+                        var meta = new MetadataBuilder();
+                        // Take all existing metadata
+                        meta.Add(partialSchema[i].Metadata, selector: s => true);
+                        // Add label names
+                        if (i == scoreColumnIndex)
+                            meta.Add(labelNameKind, labelNameType, labelNameGetter);
```

Comment:
```   if (col == _scoreCol && kind == _metadataKind)
                        {
                            _labelNameGetter.Marshal(0, ref value);
                            return;
                        }
```
this is from original code.

make same if statement and if it's not true then just copy metadata from source.
if it's true, modify your selector for everything is true except metadatakind, and add you labelNameGetter to metadata.

Reply:
Ok. Just pushed a fix. Please let me know if I did that correctly.